tests/test_descriptor_file_parser.py

```diff
@@ -0,0 +1,281 @@
+            events=[
+                PreservationEvent(
+                    identifier="e01727d0-b4d9-47a5-925a-4018f9cac6b8",
+                    type="ingest",
+                    datetime=datetime(1983, 5, 17, 11, 9, 45, tzinfo=UTC),
+                    detail="Girl voice lot another blue nearly.",
+                    agent=Agent(
+                        address="matthew24@example.net", role="collection manager"
+                    ),
+                )
+            ],
+            metadata_files=[
+                FileMetadata(
+                    id="_0193972b-e591-7e28-b8cb-1babed52f606",
+                    use="DESCRIPTIVE/COMMON",
+                    ref=FileReference(
+                        locref="../metadata/00000000-0000-0000-0000-000000000001.common.json",
+                        mdtype="DOR:SCHEMA",
+                        mimetype="application/json",
+                    ),
+                ),
+                FileMetadata(
+                    id="_0193972b-e592-7647-8e51-10db514433f7",
+                    use="DESCRIPTIVE",
+                    ref=FileReference(
+                        locref="../metadata/00000000-0000-0000-0000-000000000001.metadata.json",
+                        mdtype="DOR:SCHEMA",
+                        mimetype="application/json",
+                    ),
+                ),
+                FileMetadata(
+                    id="RIGHTS1",
+                    use="RIGHTS",
+                    ref=FileReference(
+                        locref="https://creativecommons.org/publicdomain/zero/1.0/",
+                        mdtype="OTHER",
+                    ),
+                ),
+            ],
+            struct_maps=[
+                StructMap(
+                    id="SM1",
+                    type=StructMapType.PHYSICAL,
+                    items=[
+                        StructMapItem(
+                            order=1,
+                            type="page",
+                            label="Page 1",
+                            asset_id="urn:dor:00000000-0000-0000-0000-000000001001",
+                        ),
+                        StructMapItem(
+                            order=2,
+                            type="page",
+                            label="Page 2",
+                            asset_id="urn:dor:00000000-0000-0000-0000-000000001002",
+                        ),
+                    ],
+                )
+            ],
+        )
+
+        parser = DescriptorFileParser(self.descriptor_path)
+        self.assertEqual(parser.get_resource(), expected_resource)
+
+    def test_parser_can_parse_asset(self):
+        descriptor_path = (
+            self.test_submission_path
+            / "xyzzy-0001-v1"
+            / "data"
+            / "00000000-0000-0000-0000-000000000001"
+            / "descriptor"
+            / "00000000-0000-0000-0000-000000001001.asset.mets2.xml"
+        )
```

Comment:
Might consider tidying this up so they both use a common `descriptor_path` and then add their specific files.

Reply:
I reduced the code here.